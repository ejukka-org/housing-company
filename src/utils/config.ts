export const adminConfig: any = {
  column1: `${process.env.REACT_APP_ADMIN_COLUMN1_HEADER_TEXT}`,
  column2: `${process.env.REACT_APP_ADMIN_COLUMN2_HEADER_TEXT}`,
  column3: `${process.env.REACT_APP_ADMIN_COLUMN3_HEADER_TEXT}`,
  column1_text: `${process.env.REACT_APP_ADMIN_COLUMN1_HEADER_TEXT2}`,
  column2_text: `${process.env.REACT_APP_ADMIN_COLUMN2_HEADER_TEXT2}`,
  column3_text: `${process.env.REACT_APP_ADMIN_COLUMN3_HEADER_TEXT2}`,
  list1: `${process.env.REACT_APP_ADMIN_LIST1}`,
  list1_text: `${process.env.REACT_APP_ADMIN_LIST1_TEXT}`
};

export const apartmentsConfig: any = {
  img1: `${process.env.REACT_APP_APARTMENTS_IMG1}`,
  img1_ALT: `${process.env.REACT_APP_APARTMENTS_IMG1_ALT}`,
  img1_text: `${process.env.REACT_APP_APARTMENTS_IMG1_TEXT}`,
  img2: `${process.env.REACT_APP_APARTMENTS_IMG2}`,
  img2_ALT: `${process.env.REACT_APP_APARTMENTS_IMG2_ALT}`,
  img2_text: `${process.env.REACT_APP_APARTMENTS_IMG2_TEXT}`,
  img3: `${process.env.REACT_APP_APARTMENTS_IMG3}`,
  img3_ALT: `${process.env.REACT_APP_APARTMENTS_IMG3_ALT}`,
  img3_text: `${process.env.REACT_APP_APARTMENTS_IMG3_TEXT}`
};

export const aboutPageConfig: any = {
  column1: `${process.env.REACT_APP_ABOUT_COLUMN1_HEADER_TEXT}`,
  column2: `${process.env.REACT_APP_ABOUT_COLUMN2_HEADER_TEXT}`,
  column3: `${process.env.REACT_APP_ABOUT_COLUMN3_HEADER_TEXT}`,
  column1_text: `${process.env.REACT_APP_ABOUT_COLUMN1_HEADER_TEXT2}`,
  column2_text: `${process.env.REACT_APP_ABOUT_COLUMN2_HEADER_TEXT2}`,
  column3_text: `${process.env.REACT_APP_ABOUT_COLUMN3_HEADER_TEXT2}`,
  list1: `${process.env.REACT_APP_ABOUT_LIST1}`,
  list1_text: `${process.env.REACT_APP_ABOUT_LIST1_TEXT}`,
  list2: `${process.env.REACT_APP_ABOUT_LIST2}`,
  list2_text: `${process.env.REACT_APP_ABOUT_LIST2_TEXT}`,
  list3: `${process.env.REACT_APP_ABOUT_LIST3}`,
  list3_text: `${process.env.REACT_APP_ABOUT_LIST3_TEXT}`,
  list4: `${process.env.REACT_APP_ABOUT_LIST4}`,
  list4_text: `${process.env.REACT_APP_ABOUT_LIST4_TEXT}`,
  list5: `${process.env.REACT_APP_ABOUT_LIST5}`,
  list5_text: `${process.env.REACT_APP_ABOUT_LIST5_TEXT}`
};

export const homePageConfig: any = {
  menuTitle1: `${process.env.REACT_APP_MENU_TITLE1}`,
  menuTitle1Link: `${process.env.REACT_APP_MENU_TITLE1_LINK}`,
  menuTitle2: `${process.env.REACT_APP_MENU_TITLE2}`,
  menuTitle2Link: `${process.env.REACT_APP_MENU_TITLE2_LINK}`
};

export const googleMapconfig = {
  latitude: `${process.env.REACT_APP_LATITUDE}`,
  longitude: `${process.env.REACT_APP_LONGITUDE}`
};

export const headerConfig = {
  headerText: `${process.env.REACT_APP_APP_headerText}`,
  headerTextAdmin: `${process.env.REACT_APP_APP_headerTextAdmin}`,
  link1: `${process.env.REACT_APP_APP_link1}`,
  link1Text: `${process.env.REACT_APP_APP_link1Text}`,
  link2: `${process.env.REACT_APP_APP_link2}`,
  link2Text: `${process.env.REACT_APP_APP_link2Text}`,
  link3: `${process.env.REACT_APP_APP_link3}`,
  link3Text: `${process.env.REACT_APP_APP_link3Text}`,
  link4: `${process.env.REACT_APP_APP_link4}`,
  link4Text: `${process.env.REACT_APP_APP_link4Text}`,
  link5: `${process.env.REACT_APP_APP_link5}`,
  link5Text: `${process.env.REACT_APP_APP_link5Text}`,
  link6: `${process.env.REACT_APP_APP_link6}`,
  link6Text: `${process.env.REACT_APP_APP_link6Text}`,
  admin_page_link: `${process.env.REACT_APP_APP_admin_page_link_text}`,
  admin_page_link_text: `${process.env.REACT_APP_APP_admin_page_link}`,

  header_text1: `${process.env.REACT_APP_APP_header_text1}`,
  header_text2: `${process.env.REACT_APP_APP_header_text2}`,
  header_text3: `${process.env.REACT_APP_APP_header_text3}`,
  header_text4: `${process.env.REACT_APP_APP_header_text4}`,
  sign_in_text: `${process.env.REACT_APP_APP_sign_in_text}`,
  sign_out_text: `${process.env.REACT_APP_APP_sign_out_text}`
};
