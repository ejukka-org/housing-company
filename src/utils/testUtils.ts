const initTest = () => {
  let container: any;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });
};

export default initTest;
