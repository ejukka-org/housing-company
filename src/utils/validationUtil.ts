const isValidAdminEmail = (email: string): boolean => {
  try {
    const filteredList = whiteList.filter(list => list === email);
    return filteredList.length > 0 ? true : false;
  } catch (error) {
    return false;
  }
};

const whiteList = ["jukka.edvardsson@gmail.com", "test@test.com"];

export default isValidAdminEmail;
