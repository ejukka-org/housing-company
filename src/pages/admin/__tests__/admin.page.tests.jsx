import React from 'react';
import AdminPage from '../admin.page';
import reduxStore from "../../../redux/store";
import { act } from "react-dom/test-utils";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import pretty from "pretty";
import initTest from "../../../utils/testUtils"
import { container } from "../../../utils/testUtils"

initTest();

test('Renders Admin page', () => {
    const store = reduxStore.store;
    act(() => {
        ReactDOM.render(
            <Provider store={store}>
                <AdminPage />
            </Provider>,
            container
        );
    });
    expect(pretty(container.innerHTML)).toMatchSnapshot();
});
