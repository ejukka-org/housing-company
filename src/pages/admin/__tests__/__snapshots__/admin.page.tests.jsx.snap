// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Admin page 1`] = `"<div>you must log in</div>"`;
